documentation/decisions/adr/2024-03-19-dashboard-storage.md

```diff
@@ -0,0 +1,89 @@
+
+[Data is hosted in S3 at $0.023 per GB-month](https://aws.amazon.com/s3/pricing/).
+
+At 0 - 6 months, S3 is a reasonable choice due to our small data sizes. Past that point, performance issues with large data sizes make S3 a non-ideal choice.
+
+At 2 - 5 years, S3's performance issues require the introduction of another query / compute layer like AWS Athena or AWS Redshift.
+
+### Redshift
+
+AWS Redshift is a Postgres-based data warehouse that you can use to store large data sets. This ADR assumes we are going to be using Redshift Serverless, rather than statically provisioned Redshift. We assume that serverless technologies will work in this case because this is a data warehouse backed by an ELT process, so it's data is not being accessed most of the time. Redshift is optimized for OLAP queries, which makes it an ideal choice for analyzing business analytics data. To query CSV data in Redshift, you must first upload it to S3 and then into Redshift, which makes Redshift feel like a "query layer" for S3.
+
+[Data is hosted in Redshift at $0.024 per GB-month](https://aws.amazon.com/redshift/pricing/), essentially the same price as S3.
+
+At 0 - 6 months, Redshift loses to S3 due to S3 being vastly easier to set up and configure.
+
+At 6 months - 2 years, Redshift starts to beat S3 as the initial setup cost has already been paid, and Redshift will start to show performance advantages relative to S3.
+
+At 2 - 5 years, performance bottle-necks with S3 mean that Redshift easily wins any comparison.
+
+### Postgres
+
+Postgres, as hosted by AWS RDS, is a SQL database that you can use to store large data sets. Many of the characteristics that apply to Redshift apply to Postgres, with the caveat that Postgres is not built for business intelligence use cases. We have an existing Postgres database, but we would not want to re-use it for our data analytics purposes.
+
+[Data is hosted in Postgres at $0.115 per GB-month](https://aws.amazon.com/rds/postgresql/pricing/), higher than S3 and Redshift.
+
+In every time range, Postgres loses to Redshift due to Postgres being an OLTP database built for real-time data processing. In the 2 - 5 year range, Postgres also loses due to its high GB-month hosting cost. There is however, evidence that Postgres is faster than Redshift, and that Redshift's OLAP advantages don't kick in until the data grows to terabtyes in size.
```

Comment:
Pros of postgres include:

* Open Source
* Keep stack simplified as we already use it
* Easier to setup locally for open source devs or local dev and for other teams to adopt 
* I'm still skeptical that we will ever need to run queries that require an OLTP database
* We could use the same database for our open source BI tool (Metabase/Superset use postgres)